ouster-ros/src/os_sensor_node.cpp

```diff
@@ -571,25 +574,49 @@ uint8_t OusterSensor::compose_config_flags(
     return config_flags;
 }
 
-void OusterSensor::configure_sensor(const std::string& hostname,
-                                    sensor::sensor_config& config) {
-    if (config.udp_dest && sensor::in_multicast(config.udp_dest.value()) &&
-        !mtp_main) {
-        if (!get_config(hostname, config, true)) {
-            RCLCPP_ERROR(get_logger(), "Error getting active config");
-        } else {
-            RCLCPP_INFO(get_logger(), "Retrived active config of sensor");
+bool OusterSensor::configure_sensor(const std::string& hostname,
+                                    sensor::sensor_config& config, bool retry) {
+
+    bool is_configured = false;
+    bool is_first_attempt = true;
+
+    do {
+        // Throttling
+        if (!is_first_attempt) {
+            if (config.udp_dest && sensor::in_multicast(config.udp_dest.value()) &&
+                !mtp_main) {
+                if (!get_config(hostname, config, true)) {
+                    RCLCPP_ERROR_THROTTLE(get_logger(), *get_clock(), THROTTLING_TIME, "Error getting active config");
+                } else {
+                    RCLCPP_INFO(get_logger(), "Retrieved active config of sensor");
+                }
+            } else {
+                try {
+                    uint8_t config_flags = compose_config_flags(config);
+                    if (!set_config(hostname, config, config_flags)) {
+                        RCLCPP_ERROR_THROTTLE(get_logger(), *get_clock(), THROTTLING_TIME,  "Error getting active config");
+                    } else {
+                        is_configured = true;
+                    }
+                } catch (const std::exception& e) {
+                     RCLCPP_ERROR_THROTTLE(get_logger(), *get_clock(), THROTTLING_TIME, "Error setting config:  %s", e.what());
+                }
+
+            }
+
         }
-        return;
-    }
+        is_first_attempt = false;
+        std::this_thread::sleep_for(std::chrono::milliseconds(500));
+    } while (!is_configured && retry);
```

Comment:
The first attempt variable here is pretty weird. If it is first try you simply just wait and if you have retry as false you actually will not get any configs.

Reply:
Good catch! Guess I blindly copied this from the other PR and might have even copied it badly idk. But I think this was broken in the current state if we set the parameter to false. Fixed it (remove the first try thing entirely) and tested with both values for the paramter, wokr s as expected